### PR TITLE
refactor(core): clear `@ts-ignore` from workflow and core packages

### PR DESCRIPTION
**Pull request by ivov**

### CODE REVIEW

**Review by netroy — DISMISSED** [review]
This is great. Thanks 👍🏽

**Comment by netroy** [comment]
Should we merge this after the fix release is out later today?
I don't see a breaking change here, so I'm okay either way. It's your call to merge it now, or on Monday.

**Comment by ivov** [comment]
I'll wait for Monday.

**Review by netroy — DISMISSED** [review]

File: packages/core/src/NodeExecuteFunctions.ts [hunk 1]
```diff
@@ -275,17 +272,16 @@ async function parseRequestObject(requestObject: IDataObject) {
...
 			const headers = axiosConfig.data.getHeaders();
 			axiosConfig.headers = Object.assign(axiosConfig.headers || {}, headers);
 			await generateContentLengthHeader(axiosConfig.data, axiosConfig.headers);
 		} else if (requestObject.body !== undefined) {
 			// If we have body and possibly form
-			if (requestObject.form !== undefined) {
+			if (requestObject.form !== undefined && requestObject.body) {
 				// merge both objects when exist.
-				// @ts-ignore
 				requestObject.body = Object.assign(requestObject.body, requestObject.form);
```
Comment (netroy) [inline packages/core/src/NodeExecuteFunctions.ts:285]: We should change this to either
`Object.assign(requestObject.body, requestObject.form)` or `requestObject.body = Object.assign({}, requestObject.body, requestObject.form)`

we should also perhaps (eventually) create a linting rule for `Object.assign` should be used.

File: packages/core/src/NodeExecuteFunctions.ts [hunk 2]
```diff
@@ -1168,20 +1179,21 @@ export async function requestOAuth1(
 		secret: oauthTokenData.oauth_token_secret as string,
 	};
 
-	// @ts-ignore
+	// @ts-expect-error @TECH_DEBT: Improve typings for requestOptions
 	requestOptions.data = { ...requestOptions.qs, ...requestOptions.form };
```
Comment (netroy) [inline packages/core/src/NodeExecuteFunctions.ts:1183]: I'd rather change this to `// @ts-expect-error @TECH_DEBT: get rid of the request library` 😅

**Review by netroy — DISMISSED** [review]
besides the `Object.assign` stuff, lgtm

File: packages/core/src/NodeExecuteFunctions.ts (hunk 1 above)
Comment (ivov) [reply packages/core/src/NodeExecuteFunctions.ts:285]: This pattern occurs very frequently elsewhere. Also okay if we fix this in full but separately through the lint rule to be written?

Comment (netroy) [reply packages/core/src/NodeExecuteFunctions.ts:285]: okay

**Review by netroy — APPROVED** [review]

**Comment by janober** [comment]
Got released with `n8n@0.201.0`

**Pull request merged**